packages/gatsby-theme-docs/src/utils/navigate-with-filters.js

```diff
@@ -0,0 +1,33 @@
+import { navigate } from '@reach/router';
+import { encode } from 'qss';
+import extractQueryParameters from './extract-query-parameters';
+
+function navigateWithFilters(filters, location) {
+  const currentQueryParameters = extractQueryParameters(location);
+  const newQueryParameters = removeUnnecessaryParameters({
+    ...currentQueryParameters,
+    ...filters,
+  });
+
+  const to =
+    Object.keys(newQueryParameters).length > 0
+      ? encode(newQueryParameters, '?')
+      : location.pathname;
```

Comment:
> I don't think there is a need to manually remove those params, the encode function does this for you.

I have 2 reasons:
* to remove the questions mark from the url.
* the encode function does not remove fields with empty strings, so when date field is cleared, you end up with something like this in the url - `http://localhost:8000/releases?fromFilterDate=`